include/VCF_Tasks.h

```diff
@@ -1,79 +1,97 @@
- * (11-100) - Performance-critical functions (reading pedals, interfacing w/ inverters, etc)
- * (100+) - Telemetry, logging functions (could be idle functions, too)
- */
-
-#ifndef VCF_TASKS
-#define VCF_TASKS
-
-/* From HT_SCHED library */
-#include "ht_sched.hpp"
-
-/* From shared_firmware_types library */
-#include "SharedFirmwareTypes.h"
-
-
-
-/**
- * The read_adc1 task will command adc1 to sample all eight channels, convert the outputs, and
- * store them in structs defined in shared_firmware_types. This function relies on adc_1 being
- * defined in VCFGlobals.h.
- */
-bool init_read_adc1_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo& taskInfo);
-bool run_read_adc1_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo& taskInfo);
-extern HT_TASK::Task read_adc1_task;
-
-
-
-/**
- * NOTE: These channels are UNUSED BY DEFAULT and exist ONLY FOR TESTING. You may edit this
- * manually to add sensors.
- * 
- * The read_adc2 task will command adc2 to sample all eight channels, convert the outputs, and
- * store them in a struct defined in shared_firmware_types. This function relies on adc_2 being
- * defined in VCFGlobals.h.
- */
-bool init_read_adc2_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo& taskInfo);
-bool run_read_adc2_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo& taskInfo);
-extern HT_TASK::Task read_adc2_task;
-
-/**
- * The buzzer_control task will control the buzzer control pin. This function relies on the 
- * buzzer_control pin definition in VCF_Constants.h;
+ * Critical functions: 1-10ms (100-1000Hz)
+ * Performance monitoring: 10-100ms (10-100Hz)
+ * Status updates, non-critical functions: 100ms+ (<10Hz)
  */
-bool init_buzzer_control_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo& taskInfo);
-bool run_buzzer_control_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo& taskInfo);
-extern HT_TASK::Task buzzer_control_task;
-
-/**
- * The read_gpio task will read the GPIO pins and store the results in a struct defined in 
- * shared_firmware_types. This function relies on the GPIO pins being defined in VCFConstants.h.
- */
-bool init_read_gpio_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo& taskInfo);
-bool run_read_gpio_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo& taskInfo);
-extern HT_TASK::Task read_gpio_task;
-
-
-
 
-#endif /* VCF_TASKS */
\ No newline at end of file
+ #ifndef VCF_TASKS
+ #define VCF_TASKS
+ 
+ 
+ /* From shared_firmware_types library */
+ #include "SharedFirmwareTypes.h"
+ 
+ #define _TASK_MICRO_RES
+ #include <TScheduler.hpp>
+ 
+ extern TsScheduler task_scheduler;
+ 
+ /**
+  * The read_adc1 task will command adc1 to sample all eight channels, convert the outputs, and
+  * store them in structs defined in shared_firmware_types. This function relies on adc_1 being
+  * defined in VCFGlobals.h.
+  */
+ bool init_read_adc1_task();
+ void run_read_adc1_task();
+ extern TsTask read_adc1_task;
```

Comment:
change to follow the paradigm in VCR where tasks are declared and assigned in main instead of having externs.